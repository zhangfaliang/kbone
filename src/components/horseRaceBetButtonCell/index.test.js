import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import HorseRaceBetButtonCell from './index';
import HorseRaceBetButton from '../horseRaceBetButton';

describe('HorseRaceBetButtonCell', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceBetButtonCell>
        <HorseRaceBetButton
          optionName="sp"
          className="test"
          optionId="2.3"
          handicap="4"
          odds="6.7"
          historySp={[]}
          bettingSlipId="65"
          playtypeName="456"
          title="456"
          betLimit="5000"
          playtypeCode="sdf"
          prefixCls="withdraw-tir"
          showOdds={true}
        />
      </HorseRaceBetButtonCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls', () => {
    const wrapper = render(
      <HorseRaceBetButtonCell prefixCls="default">
        <HorseRaceBetButton
          optionName="sp"
          className="test"
          optionId="2.3"
          handicap="4"
          odds="6.7"
          historySp={[]}
          bettingSlipId="65"
          playtypeName="456"
          title="456"
          betLimit="5000"
          playtypeCode="sdf"
          prefixCls="withdraw-tir"
          showOdds={true}
        />
      </HorseRaceBetButtonCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('selectBetBtn', () => {
    const wrapper = mount(
      <HorseRaceBetButtonCell prefixCls="default">
        <HorseRaceBetButton
          optionName="sp"
          className="test"
          optionId="2.3"
          handicap="4"
          odds="6.7"
          historySp={[]}
          bettingSlipId="65"
          playtypeName="456"
          title="456"
          betLimit="5000"
          playtypeCode="sdf"
          prefixCls="withdraw-tir"
          showOdds={true}
        />
      </HorseRaceBetButtonCell>
    );
    const selectBetBtn = wrapper.prop('selectBetBtn');
    const click = jest.fn(selectBetBtn);
    wrapper
      .find('HorseRaceBetButtonCell')
      .find('HorseRaceBetButton')
      .simulate('click');
    expect(click()).toBe('selectBetBtn');
  });
  it('HorseRaceBetButtonCell', () => {
    const selectBetBtn = () => 'selectBetBtn';
    const click = jest.fn(selectBetBtn);
    const wrapper = render(
      <HorseRaceBetButtonCell prefixCls="default" selectBetBtn={click}>
        <HorseRaceBetButtonCell />
      </HorseRaceBetButtonCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
