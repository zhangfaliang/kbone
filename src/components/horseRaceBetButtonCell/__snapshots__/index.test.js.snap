// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorseRaceBetButtonCell HorseRaceBetButtonCell 1`] = `
<div
  class="default-horse-race-bet-btn-cell"
/>
`;

exports[`HorseRaceBetButtonCell base 1`] = `
<div
  class="default-horse-race-bet-btn-cell"
>
  <div
    class="test withdraw-tir-horse-race-bet-btn"
  >
    <span
      class="odds"
    >
      6.7
    </span>
  </div>
</div>
`;

exports[`HorseRaceBetButtonCell prefixCls 1`] = `
<div
  class="default-horse-race-bet-btn-cell"
>
  <div
    class="test withdraw-tir-horse-race-bet-btn"
  >
    <span
      class="odds"
    >
      6.7
    </span>
  </div>
</div>
`;
